### ISSUE TITLE
Filtering by list entry match

### ISSUE THREAD

**Issue by LiNengRan** [dgraph-io/dgraph#3787]
If you suspect this could be a bug, follow the template.

- What version of Dgraph are you using?
Latest

- Have you tried reproducing the issue with latest release?
Yes

- What is the hardware spec (RAM, OS)?
OSX High Sierra 156 G

- Steps to reproduce the issue (command/config used to run Dgraph).
Ratel localhost 8000

- Expected behaviour and actual result.
Consistently can ONLY match the FINAL  item in a filter over an indexed list. Using data from another issue on here, set up with mutation as below:
```
{
  "set": [
    {
      "Nodes": [
        {
          "Node_Name": "Node_01",
          "timestamps": [
            "2018-01-01",
            "2018-02-01",
            "2018-03-01",
            "2018-04-01"
          ]
        },
        {
          "Node_Name": "Node_02",
          "timestamps": [
            "2019-01-01",
            "2019-02-01",
            "2019-03-01",
            "2019-04-01"
          ]
        },
         {
          "Node_Name": "Node_03",
          "timestamps": [
            "2018-05-01",
            "2018-04-01"
          ]
        },
        {
          "Node_Name": "Node_04",
          "timestamps": [
            "2019-06-01"
          ]
        },
        {
          "Node_Name": "Node_05",
          "timestamps": [
            "2017-06-01"
          ]
        }
      ]
    }
  ]
}

```
Now index the timesstamps as datettime list:
`<timestamps>: [datetime] @index(month) .`

Now try this query and everything will be fine:
```
{
  query(func: has(Nodes))  {
    Nodes @filter(eq(timestamps, "2018-04-01") ){ expand(_all_)}
  }
  }

{
```
```

"data": {
"query": [
{
"Nodes": [
{
"Node_Name": "Node_01",
"timestamps": [
"2018-04-01T00:00:00Z"
]
},
{
"Node_Name": "Node_03",
"timestamps": [
"2018-04-01T00:00:00Z"
]
}
]
}
]
}
}

```
 @LiNengRan

BUT now just make 1 small change, change the month in the query search string to be '03' instead of '04' (note that 03 is the month of a legitimate timestamp, of Node 01:
"2018-03-01",
```

{
  query(func: has(Nodes))  {
    Nodes @filter(eq(timestamps, "2018-03-01") ){ expand(_all_)}
  }
}
```
Now no data is returned at all:

```
 "data": {
    "query": []
  }

```
This is not only datetime lists, I've had the same exact thing happen with ordinary lists of strings that are term indexed. 

Bottom Line: IT ONLY KNOWS HOW TO MATCH THE FINAL LIST ENTRY. 

Please fix, thank you!

**Comment by LiNengRan** [comment]
Sorry the "good" return above (last list entry queried for) looks like this:

{

  "data": {
    "query": [
      {
        "Nodes": [
          {
            "Node_Name": "Node_01",
            "timestamps": [
              "2018-04-01T00:00:00Z"
            ]
          },
          {
            "Node_Name": "Node_03",
            "timestamps": [
              "2018-04-01T00:00:00Z"
            ]
          }
        ]
      }
    ]
  }
}

**Comment by MichelDiz** [comment]
Hey,

I found the issue. You need to set the Schema before mutate. If you mutate before setting the schema, the mutation you lost the values. That's the only issue here.

If you set the schema you're good to go.

I'll close this issue and open a proper one for the case. But you for you case right now, just set the schema before any mutation.

**Comment by LiNengRan** [comment]
Hey thanks 1 million! You guys are great!!

-S
